lib/modules/manager/pipenv/extract.ts

```diff
@@ -146,10 +175,10 @@ export async function extractPackageFile(
     extractedConstraints.python = `== ${pipfile.requires.python_full_version}`;
   }
 
-  if (is.nonEmptyString(pipfile.packages?.pipenv)) {
-    extractedConstraints.pipenv = pipfile.packages.pipenv;
-  } else if (is.nonEmptyString(pipfile['dev-packages']?.pipenv)) {
-    extractedConstraints.pipenv = pipfile['dev-packages'].pipenv;
+  const pipenv_constraints = categories.find(([, section]) => section?.pipenv);
+
+  if (pipenv_constraints) {
+    extractedConstraints.pipenv = pipenv_constraints[1].pipenv;
```

Comment:
why only the first? shouldn't we include all constraints?

Reply:
This is matching the previous logic where it would ignore constraints in the `dev-packages` section if it was found in the `packages` section. I'm not sure how Pipenv itself would resolve this, so left it as-is.

That said, there is a bug here: pipenv_constraints is badly named and only contains one "thing".